pandas/tests/test_nanops.py

```diff
@@ -1005,6 +1005,39 @@ def prng(self):
         return np.random.RandomState(1234)
 
 
+class TestNumpyNaNFunctions(object):
+
+    # xref GH 19629
+    def setup_method(self, method):
+        self.test_series = pd.Series([1, 2, 3, 4, 5, 6])
+        self.test_df = pd.DataFrame([[1, 2, 3], [4, 5, 6]])
+
+    def compare_method_output(self, data, method, nan_method):
+        tm.assert_almost_equal(data.agg(method),
+                               data.agg(nan_method),
+                               check_exact=True)
+
+    @pytest.mark.parametrize("standard, nan_method", [
```

Comment:
u can parametrize over df and series (or make into fixtures)

then can inline the compare function

Reply:
I'm not sure on a good way to handle the np compat tests if we took this approach, I'd be open to suggestions though

Reply:
```
@pytest.fixture(params=[Series(....), DataFrame()])
def obj(request):
    return request.param
```